app/controllers/spree/khalti_controller.rb

```diff
@@ -0,0 +1,116 @@
+module Spree
+  class KhaltiController < StoreController
+    skip_before_action :verify_authenticity_token
+
+    def payment
+      payment_method = PaymentMethod.find(params[:payment_method_id])
+
+      order = current_order || raise(ActiveRecord::RecordNotFound)
+
+      current_payment = create_payment(payment_method)
+      current_payment.pend!
+
+      begin
+        # API call to khalti
+        headers = {
```

Comment:
Shouldn't we be making call from a service which connects with Khalti API? @poudelprakash your take on this?